test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -51,11 +57,86 @@
+          jruby-service (reify jruby/JRubyPuppetService
+                          (borrow-instance [_ _] jruby-instance)
+                          (return-instance [_ jruby _]
+                            (reset! returned-jruby jruby))
+                          (get-environment-class-info [_ jruby env]
+                            (reset! get-env-jruby jruby)
+                            (if (= env "production")
+                              {"/one/file"
+                               [
+                                {"name" "oneclass",
+                                 "params" [
+                                           {"name" "oneparam",
+                                            "type" "String",
+                                            "default_literal" "'literal'",
+                                            "default_source" "literal"},
+                                           {"name" "twoparam",
+                                            "type" "Integer",
+                                            "default_literal" "3",
+                                            "default_source" "3"}]
+                                 },
+                                {"name" "twoclass"
+                                 "params" []}],
+                               "/two/file" []})))
+          handler (fn ([req] {:request req}))
+          app (build-ring-handler handler "1.2.3" jruby-service)
+          request (partial app-request app)]
+      (testing "returns good data for environment that exists"
+        (let [response (request
+                        "/v3/environment_classes?environment=production")
+              body-from-json (cheshire/parse-string (:body response))]
+          (is (= 200 (:status response))
+              "Unexpected status code returned from request")
+          (is (= {"name" "production",
+                  "files" [
+                           {"path" "/one/file",
+                            "classes" [{
+                                        "name" "oneclass"
+                                        "params" [
+                                                  {"name" "oneparam",
+                                                   "type" "String",
+                                                   "default_literal" "'literal'",
+                                                   "default_source" "literal"},
+                                                  {"name" "twoparam",
+                                                   "type" "Integer",
+                                                   "default_literal" "3",
+                                                   "default_source" "3"}]},
+                                       {
+                                        "name" "twoclass"
+                                        "params" []}]},
+                           {"path" "/two/file"
+                            "classes" []}]}
+                 body-from-json)
```

Comment:
In 2dd5530, I kept the assertion that a 200 is returned for a request with a good environment.  I had to keep a bit of the mocking for that - at least a JRubyPuppetService that can return data for what it thinks is a good environment.  I did take out some of the assertions and state around the use of JRuby pool instances and removed the large payload that was being translated previously since this is arguably already covered sufficiently in the integration tests.